chsdi/views/files.py

```diff
@@ -0,0 +1,242 @@
+    #table = get_table(table_name='geoadmin-file-storage')
+
+    return table
+
+
+    # return get_table(table_name='geoadmin-file-storage')
+
+def _add_item(id, file_id=False):
+    table = _get_dynamodb_table()
+    try:
+        table.put_item(
+            data={
+                'adminId': id,
+                'fileId': file_id,
+                'timestamp': time.strftime('%Y-%m-%d %X', time.localtime())
+            }
+        )
+    except Exception as e:
+            raise exc.HTTPBadRequest('Error during put item %s' % e)
+    return True
+
+
+def _is_admin_id(adminId):
+    # TODO: search in database if it's an adminId
+    table = _get_dynamodb_table()
+    try:
+        table.get_item(adminId=str(adminId))
+    except ItemNotFound:
+        return False
+
+    return True
+
+
+def _get_file_id_from_admin_id(admin_id):
+    # TODO: get the file id from database
+    fileId = None
+    table = _get_dynamodb_table()
+    try:
+        item = table.get_item(adminId=str(admin_id))
+        fileId = item.get('fileId')
+    except Exception as e:
+        raise exc.HTTPBadRequest('The id %s doesn\'t exist. Error is: %s' % (id, e))
+    if fileId is None:
+        return False
+    return fileId
+
+
+@view_defaults(renderer='jsonp', route_name='files')
+class FileView(object):
+
+    def __init__(self, request):
+        self.request = request
+        self.bucket = self._get_bucket()
+        if request.matched_route.name == 'files':
+            self.admin_id = None
+            id = request.matchdict['id']
+            if _is_admin_id(id):
+                self.admin_id = id
+                self.file_id = _get_file_id_from_admin_id(self.admin_id)
+            else:
+                self.file_id = id
+            try:
+                key = self.bucket.get_key(self.file_id)
+                if key is not None:
+                    self.key = key
+                else:
+                    raise exc.HTTPNotFound('File %s not found' % self.file_id)
+            except:
+                raise exc.HTTPInternalServerError('Cannot access file with id=%s' % self.file_id)
+
+    def _get_uuid(self):
+        return base64.urlsafe_b64encode(uuid.uuid4().bytes).replace('=', '')
+
+    def _get_bucket(self):
+        # TODO use profile instead when correctly installed
+        PROFILE_NAME = 'geoadmin-filestorage'
+        BUCKET_NAME = self.request.registry.settings['geoadmin_file_storage_bucket']
+        config = ConfigParser.ConfigParser()
+        config.read(["/home/ltmom/.boto"])
+
+        access_key = config.get(PROFILE_NAME, 'aws_access_key_id')
+        secret_key = config.get(PROFILE_NAME, 'aws_secret_access_key')
+        try:
+            # TODO use this
+            #conn = S3Connection(profile_name='wmts_test')
+            conn = S3Connection(aws_access_key_id=access_key, aws_secret_access_key=secret_key)
+            bucket = conn.get_bucket(BUCKET_NAME)
+        except Exception as e:
```

Comment:
to be more accurate.
If the bucket does not exist, an S3ResponseError will be raised.